torch/generator.py

```diff
@@ -0,0 +1,106 @@
+import torch
+
+
+class Generator(torch._C._GeneratorBase):
+    r"""Generator(device='cpu', default=False) -> Generator
+
+    Creates and returns a generator object which manages the state of the algorithm that
+    produces pseudo random numbers. Used as a keyword argument in many random tensors in
+    :ref:`inplace-random-sampling`. Currently only creation of CPU Generator is supported through
+    this API.
+
+    Arguments:
+        device (:class:`torch.device`, optional): the desired device for the generator.
+        default (bool, optional): If using the default CPU/CUDA generator.
```

Comment:
This can be written more clearly. Basically, setting `default=True` means you "construct" the default generator, which will coincide with any other constructions of the default generator.

I am a little perplexed; is this true? `torch.Generator(default=True) == torch.Generator(default=True)`? I don't see how it could be, but it seems like a desirable property to have.

Reply:
Hot take: you shouldn't make `default` a constructor argument. Instead, at initialization time in the Python bindings, you should make a single distinguished default generator (making it the correct class by poking out the class `torch.generator.Generator`) and then have some function to get at it. Then object identity desired above will hold.